Backport missing cluster health fixes from ES

seut opened the pull request:
## Summary of the changes / Why this improves CrateDB

See commits.

This should solve flaky tests which asserts cluster health and shard allocations by waiting for a green cluster health using `ensureGreen`. E.g. the random failures of the `FilteringAllocationIT` should be solved now, the last commit removes the assertBusy calls in there.

mfussenegger commented on server/src/test/java/org/elasticsearch/gateway/GatewayIndexStateIT.java:
```diff
@@ -0,0 +1,585 @@
+    }
+
+    @Test
+    @Ignore("Creating a table without any data node will take very long as internally at CrateDB, a table creation" +
```
Could we use the slow annotation here instead, or does it currently fail?

seut replied:
Was not sure if it's worth and maybe adds another flakiness. I've added a fixup.

mfussenegger reviewed (commented):
Left a question, otherwise looks good :+1:

mfussenegger commented:
There is btw. still a diff

![image](https://user-images.githubusercontent.com/38700/100374632-e08cd980-300c-11eb-8419-291ec62e2f5f.png)

We've a `masterOperation` overload that got removed in ES - worth following up on?

seut commented:
@mfussenegger 

> There is btw. still a diff

I've removed the overload (sorry I've already rebased and eliminated the dedicated fixup). Or did you mean it in a different way?

mfussenegger commented:
>  Or did you mean it in a different way?

No, all good :+1:
